state/tx/indexer/kv.go

```diff
@@ -0,0 +1,55 @@
+package indexer
+
+import (
+	"bytes"
+	"fmt"
+
+	db "github.com/tendermint/go-db"
+	"github.com/tendermint/go-wire"
+	"github.com/tendermint/tendermint/types"
+)
+
+// KV is a simplest possible indexer, backed by Key-Value storage (levelDB).
+// It could only index transaction by its identifier.
+type KV struct {
```

Comment:
This naming doesn't seem right.  Of course Golang encourages you to have the module name as part of an exported thing's name, e.g. json.Encoder rather than json.JSONEncoder, but in this case, the struct isn't a KV, it's an Indexer.  So unless we're going to have a separate package named 'tendermint/indexer/kv', we should call this KVIndexer.